compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -437,15 +544,48 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       if (shouldCheckLiterals)
         checkJSNameLiteral(sym)
 
+      if (enclosingOwner is OwnerKind.JSClass) {
+        // Private methods cannot be overloaded
+        if (sym.isMethod && isPrivateMaybeWithin(sym)) {
+          val alts = sym.owner.info.member(sym.name).filter(_.isMethod)
+          if (alts.isOverloaded) {
+            reporter.error(tree.pos,
+                "Private methods in Scala.js-defined JS classes cannot be " +
+                "overloaded. Use different names instead.")
+          }
+        }
+
+        // Cannot override a private method (even with a protected/public one)
+        sym.allOverriddenSymbols.find(isPrivateMaybeWithin).foreach { overriddenSym =>
```

Comment:
I found a much better idea: force `private[Scope]` methods to be `final` :-p And it is much easier to explain as a spec. Note that `private` and `private[this]` need not be final because they are implicitly so.